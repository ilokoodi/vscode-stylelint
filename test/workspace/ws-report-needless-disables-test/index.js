'use strict';

const path = require('path');

const pWaitFor = require('p-wait-for');
const { workspace, window } = require('vscode');

const { normalizeDiagnostic, getStylelintDiagnostics } = require('../utils');

describe('vscode-stylelint with "stylelint.reportNeedlessDisables"', () => {
	it('should work if "stylelint.reportNeedlessDisables" is enabled', async () => {
		// Open the './test.css' file.
		const cssDocument = await workspace.openTextDocument(path.resolve(__dirname, 'test.css'));

		await window.showTextDocument(cssDocument);

		// Wait for diagnostics result.
		await pWaitFor(() => getStylelintDiagnostics(cssDocument.uri).length > 0, { timeout: 5000 });

		const diagnostics = getStylelintDiagnostics(cssDocument.uri);

		expect(diagnostics.map(normalizeDiagnostic)).toMatchSnapshot();
	});
});
