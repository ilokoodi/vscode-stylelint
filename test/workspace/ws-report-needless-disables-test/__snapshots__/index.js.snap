// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vscode-stylelint with "stylelint.reportNeedlessDisables" should work if "stylelint.reportNeedlessDisables" is enabled 1`] = `
Array [
  Object {
    "code": "indentation",
    "message": "unused rule: indentation, start line: 4, end line: 4",
    "range": Object {
      "end": Object {
        "character": 15,
        "line": 3,
      },
      "start": Object {
        "character": 0,
        "line": 3,
      },
    },
    "severity": 1,
    "source": "stylelint",
  },
  Object {
    "code": "indentation",
    "message": "unused rule: indentation, start line: 7, end line: 11",
    "range": Object {
      "end": Object {
        "character": 34,
        "line": 10,
      },
      "start": Object {
        "character": 0,
        "line": 6,
      },
    },
    "severity": 1,
    "source": "stylelint",
  },
  Object {
    "code": "indentation",
    "message": "unused rule: indentation, start line: 15, end line: 15",
    "range": Object {
      "end": Object {
        "character": 56,
        "line": 14,
      },
      "start": Object {
        "character": 0,
        "line": 14,
      },
    },
    "severity": 1,
    "source": "stylelint",
  },
  Object {
    "code": "indentation",
    "message": "unused rule: indentation, start line: 18",
    "range": Object {
      "end": Object {
        "character": 0,
        "line": 21,
      },
      "start": Object {
        "character": 0,
        "line": 17,
      },
    },
    "severity": 1,
    "source": "stylelint",
  },
  Object {
    "code": Object {
      "target": Object {
        "authority": "stylelint.io",
        "path": "/user-guide/rules/indentation",
        "scheme": "https",
      },
      "value": "indentation",
    },
    "message": "Expected indentation of 4 spaces (indentation)",
    "range": Object {
      "end": Object {
        "character": 2,
        "line": 2,
      },
      "start": Object {
        "character": 2,
        "line": 2,
      },
    },
    "severity": 0,
    "source": "stylelint",
  },
]
`;
